src/main/java/teammates/storage/sqlapi/NotificationsDb.java

```diff
@@ -35,7 +37,7 @@ public Notification createNotification(Notification notification)
             throw new InvalidParametersException(notification.getInvalidityInfo());
         }
 
-        if (getNotification(notification.getNotificationId().toString()) != null) {
+        if (notification.getNotificationId() != null && getNotification(notification.getNotificationId()) != null) {
             throw new EntityAlreadyExistsException(String.format(ERROR_CREATE_ENTITY_ALREADY_EXISTS,
```

Comment:
At this point I think `notification.getNotificationId()` will always return `null` as the ID generation only happens when session is flushed, which means this block of code will never be run at all.